filter_plugins/ec2_custom.py

```diff
@@ -130,12 +140,16 @@ def get_sg_id_result(result_list):
 
 def get_launch_configs(launch_configs, stack_name):
     import json
+    import re
+
     configs = []
     launch_configs = json.loads(launch_configs)
 
     if 'LaunchConfigurations' in launch_configs:
+        component_regex = "({})".format('|'.join(VALID_APP_COMPONENT_NAMES))
         for launch_config in launch_configs['LaunchConfigurations']:
-            if stack_name in launch_config['LaunchConfigurationName']:
+            pattern = re.compile("^{}-{}$".format(component_regex, stack_name))
+            if pattern.match(launch_config['LaunchConfigurationName']):
```

Comment:
the launch config name includes a hash suffix to ensure uniqueness, so this would not match with the `$` on L151